spec/consensus/proposer-based-timestamp/pbts_001_draft.md

```diff
@@ -0,0 +1,268 @@
+`2f + 1` correct validators.
+   1. If there are `n` different timestamps in the  `precommit` messages, the proposer can use any subset of timestamps that add up to `2f + 1` 
+   of the voting power in order to compute the median.
+   1. If the validators decide in different rounds, the proposer can decide on which round the median computation is based. 
+1. **Liveness.** The liveness of the protocol:
+   1. does not depend on clock synchronization,
+   1. depends on bounded message delays.
+1. **Relation to real time.** There is no clock synchronizaton, which implies that there is **no relation** between the computed block `time` and real time.
+1. **Aggregate signatures.** As the `precommit` messages contain the local times, all these `precommit` messages typically differ in the time field, which **prevents** the use of aggregate signatures.
+
+## Suggested Proposer-Based Time
+
+### Outline
+
+An alternative approach to time has been discussed: Rather than having the validators send the time in the `precommit` messages, the proposer in the consensus algorithm sends its time in the `propose` message, and the validators locally check whether the time is OK (by comparing to their local clock). 
+
+This proposed solution adds the requirement of having synchronized clocks, and other implicit assumptions.
+
+### Comparison of the Suggested Method to the Old One
+
+1. **Fault tolerance.** Maintained in the suggested protocol. 
+2. **Effect of faulty validators.** Eliminated in the suggested protocol, 
+that is, the block `time` can be corrupted only in the extreme case when 
+`>2/3` of the validators are faulty.
+1. **Proposer influence on block time.** The proposer of the next block
+has less freedom when choosing the block time.
+   1. This scenario is eliminated in the suggested protocol, provided that there are `<1/3` faulty validators.
+   1. This scenario is still there.
+1. **Liveness.** The liveness of the suggested protocol:
+   1. depends on the introduced assumptions on synchronized clocks (see below),
+   1. still depends on the message delays (unavoidable).
+1. **Relation to real time.** We formalize clock synchronization, and obtain a **well-defined relation** between the block `time` and real time.
+1. **Aggregate signatures.** The `precommit` messages free of time, which **allows** for aggregate signatures.
+
+### Protocol Overview
+
+#### Proposed Time
+We assume that the field `proposal` in the `PROPOSE` message is a pair `(v, time)`, of the proposed consensus value `v` and the proposed time `time`. 
+
+#### Reception Step
+In the reception step at node `p` at local time `now_p`, upon receiving a message `m`:
+- **if** if the message `m` is of type `PROPOSE` and satisfies `now_p - PRECISION <  m.time < now_p + PRECISION + MSGDELAY`, then mark the message as `timely`.  
```

Comment:
Should this not be `now_p - PRECISION <= m.time <= ...`?

Reply:
@istoilkovska, I guess we should check in the TLA+ spec whether we have `<` or `<=`. For safety `<` is more conservative and safe, but for the precise liveness requirements we should double-check

Reply:
After reading a comment below, I remembered that the strict inequality derives from the strict inequality used in the definition of `PRECISION`. We might change both if this seems more desirable...

Reply:
The TLA+ spec is consistent with the English spec w.r.t. the inequalities. Since there are no strong reasons to have non-strict inequalities, we should keep the strict ones.